### PR TITLE
Introduce Goleak

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR introduces the use of [goleak](https://github.com/uber-go/goleak) for detecting leakage of goroutines.
`Goleak` is added in the simplest way. This may be changed in the future. Simulation is also out of scope for now.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ clean:
 
 .PHONY: test
 test:
-	go test -short -v -shuffle=on -count=1 -race -timeout 20m ./...
```
Comment (sergefdrv) [inline Makefile:26]: This [enables non-sim tests](https://github.com/filecoin-project/mir/blob/6ed94fdef106ad3f36a5ced0d149bacfaebb094d/pkg/systems/smr/smr_test.go#L213-L216).

Comment (dnkolegov) [reply Makefile:26]: Fixed - https://github.com/filecoin-project/mir/pull/290/commits/25c8f7d27247f34a68ad7d61ee4c8930294712c9

**Review by matejpavlovic — APPROVED** [review]
👍 Thanks!

File: node_test.go [hunk 2]
```diff
@@ -179,29 +186,49 @@ type blabber struct {
...
 	return &blabber{
 		flood:     make(chan *events.EventList),
 		batchSize: batchSize,
 		period:    period,
+		stop:      make(chan struct{}),
 	}
 }
 
 // Go starts babbling, i.e., producing batches of dummy events at the rate configured at instantiation.
 // Once started, the babbler is unstoppable and keeps babbling forever.
```
[the remark this answers is not in the document]
Comment (dnkolegov) [reply node_test.go]: Fixed

File: Makefile (hunk 1 above)
Comment (dnkolegov) [reply Makefile:26]: @sergefdrv please review the fix. I just reverted the change.

**Pull request merged**